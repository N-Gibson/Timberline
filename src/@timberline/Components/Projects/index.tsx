import React from 'react'
import { Link } from 'react-router-dom'

import purpleAster from '@timberline/assets/Purple_Aster.jpg'
import venadoFront from '@timberline/assets/Venado_front.jpg'
import venadoKitchen from '@timberline/assets/Venado_Kitchen.jpg'

import './Projects.scss'

interface ProjectsProps {}

const Projects: React.FC<ProjectsProps> = () => {
  return (
    <section className="projects">
      <Link to="/projects/1">
        <div className="project_container">
          <img
            id="purple_aster"
            className="projects_photos"
            src={purpleAster}
            alt="project completed by Timberline Inc"
          />
          <label htmlFor="purple_aster">Purple Aster Spec Home</label>
        </div>
      </Link>
      <Link to="/projects/2">
        <div className="project_container">
          <img
            id="venado_front"
            className="projects_photos"
            src={venadoFront}
            alt="project completed by Timberline Inc"
          />
          <label htmlFor="venado_front">Venado Spec Home</label>
        </div>
      </Link>
      <Link to="/projects/3">
        <div className="project_container">
          <img
            id="venado_kitchen"
            className="projects_photos"
            src={venadoKitchen}
            alt="project completed by Timberline Inc"
          />
          <label htmlFor="venado_kitchen">Custom Kitchen Remodel</label>
        </div>
      </Link>
    </section>
  )
}

export { Projects }
