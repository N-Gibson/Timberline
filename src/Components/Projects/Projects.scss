.projects {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 15px;
  height: 90vh;
  padding: 180px 1rem;
  justify-items: center;
  max-width: 1440px;
  margin: 0 auto;
}

.projects_photos {
  height: 200px;
  margin: 5px;
  &:hover {
    cursor: pointer;
    transform: scale(1.05);
  }
}

.project_container {
  display: flex;
  flex-direction: column;
}
