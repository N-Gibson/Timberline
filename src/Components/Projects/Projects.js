import React from 'react';
import './Projects.scss';
import purpleAster from '../../assets/Purple_Aster.jpg';
import venadoFront from '../../assets/Venado_front.jpg';
import venadoKitchen from '../../assets/Venado_Kitchen.jpg';

export const Projects = () => {
  return (
    <section className="projects">
      <img id="purple_aster" src={purpleAster} />
      <label htmlFor="purple_aster">Purple Aster Spec Home</label>
      <img id="venado_front" src={venadoFront} />
      <label htmlFor="venado_front">Venado Spec Home</label>
      <img id="venado_kitchen" src={venadoKitchen}/>
      <label htmlFor="venado_kitchen">Custom Kitchen Remodel</label>
    </section>
  )
}

export default Projects;
