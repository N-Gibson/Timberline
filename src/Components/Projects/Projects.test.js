import React from 'react';
import { shallow } from 'enzyme';
import Projects from './Projects';

describe('Projects Component', () => {
  const wrapper = shallow(<Projects />)

  it('Projects component should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})