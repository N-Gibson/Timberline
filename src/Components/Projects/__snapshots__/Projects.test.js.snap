// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Component Projects component should match snapshot 1`] = `
<Layout
  active="projects"
>
  <section
    className="projects"
  >
    <Link
      to="/projects/1"
    >
      <div
        className="project_container"
      >
        <img
          className="projects_photos"
          id="purple_aster"
          src="Purple_Aster.jpg"
        />
        <label
          htmlFor="purple_aster"
        >
          Purple Aster Spec Home
        </label>
      </div>
    </Link>
    <Link
      to="/projects/2"
    >
      <div
        className="project_container"
      >
        <img
          className="projects_photos"
          id="venado_front"
          src="Venado_front.jpg"
        />
        <label
          htmlFor="venado_front"
        >
          Venado Spec Home
        </label>
      </div>
    </Link>
    <Link
      to="/projects/3"
    >
      <div
        className="project_container"
      >
        <img
          className="projects_photos"
          id="venado_kitchen"
          src="Venado_Kitchen.jpg"
        />
        <label
          htmlFor="venado_kitchen"
        >
          Custom Kitchen Remodel
        </label>
      </div>
    </Link>
  </section>
</Layout>
`;
